protocols/request-response/src/handler.rs

```diff
@@ -384,19 +402,47 @@ where
         loop {
             match self.worker_streams.poll_unpin(cx) {
                 Poll::Ready((_, Ok(Ok(event)))) => {
-                    return Poll::Ready(ConnectionHandlerEvent::NotifyBehaviour(event))
+                    return Poll::Ready(ConnectionHandlerEvent::NotifyBehaviour(event));
                 }
                 Poll::Ready((id, Ok(Err(e)))) => {
-                    log::debug!("Stream for request {id} failed: {e}");
+                    let event = match id {
+                        RequestId::Inbound(id) => {
+                            log::debug!("Stream for inbound request {id} failed: {e}");
+                            Event::InboundStreamFailed {
```

Comment:
I don't think we need to log _and_ return at the same time.

Reply:
done